hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerIntegrationTest.java

```diff
@@ -0,0 +1,250 @@
+                .build();
+    }
+
+    @Test
+    void successWithNoQueryParamsShardRealmNumAccountId() {
+        var allowance1 = domainBuilder.nftAllowance().persist();
+        var allowance2 = domainBuilder
+                .nftAllowance()
+                .customize(nfta -> nfta.owner(allowance1.getOwner()))
+                .persist();
+        Collection<NftAllowance> collection = List.of(allowance1, allowance2);
+
+        var result = restClient
+                .get()
+                .uri(callUri, allowance1.getOwner())
+                .accept(MediaType.ALL)
+                .retrieve()
+                .body(NftAllowancesResponse.class);
+        assertThat(result.getAllowances()).isEqualTo(mapper.map(collection));
+        assertNull(result.getLinks().getNext());
+    }
+
+    @Test
+    void successWithAllQueryParamsOrderAsc() {
+        // Creating nft allowances
+        var allowance1 = domainBuilder.nftAllowance().persist();
+        domainBuilder
+                .nftAllowance()
+                .customize(nfta -> nfta.owner(allowance1.getOwner()))
+                .persist();
+
+        // Setting up the url params
+        var uriParams = "?account.id=gte:0.0.1000&owner=true&token.id=gt:0.0.1000&limit=1&order=asc";
+
+        var nextLink =
+                "/api/v1/accounts/%s/allowances/nfts?owner=true&limit=1&order=asc&account.id=gte:%s&token.id=gt:%s"
+                        .formatted(
+                                allowance1.getOwner(),
+                                EntityId.of(allowance1.getSpender()),
+                                EntityId.of(allowance1.getTokenId()));
+
+        // Performing the GET operation
+        var result = restClient
+                .get()
+                .uri(uriParams, allowance1.getOwner())
+                .retrieve()
+                .body(NftAllowancesResponse.class);
+
+        assertThat(result.getAllowances()).isEqualTo(mapper.map(List.of(allowance1)));
+        assertThat(result.getLinks().getNext()).isEqualTo(nextLink);
+    }
+
+    @Test
+    void successWithNoOperators() {
+        // Creating nft allowances
+        var allowance1 = domainBuilder.nftAllowance().persist();
+        var allowance2 = domainBuilder
+                .nftAllowance()
+                .customize(nfta -> nfta.owner(allowance1.getOwner()))
+                .persist();
+
+        var uriParams = "?account.id={account.id}&limit=1&order=asc";
+        var nextLink = "/api/v1/accounts/%s/allowances/nfts?limit=1&order=asc&account.id=gte:%s&token.id=gt:%s"
+                .formatted(
+                        allowance1.getOwner(),
+                        EntityId.of(allowance2.getSpender()),
+                        EntityId.of(allowance2.getTokenId()));
+
+        // Performing the GET operation
+        var result = restClient
+                .get()
+                .uri(uriParams, allowance1.getOwner(), EntityId.of(allowance1.getSpender()))
```

Comment:
nit: Could pull `.get()` up to setup.

```suggestion
        var result = restClient.uri(uriParams, allowance1.getOwner(), EntityId.of(allowance1.getSpender()))
```

Reply:
Don't believe this can be done. The client needs a method.